fsspec/implementations/prefix.py

```diff
@@ -0,0 +1,196 @@
+from typing import Any, Iterable, Sequence, Union
+
+import fsspec
+from fsspec import AbstractFileSystem
+from fsspec.core import split_protocol
+from fsspec.spec import AbstractBufferedFile
+from fsspec.utils import stringify_path
+
+
+class PrefixBufferedFile(AbstractBufferedFile):
+    def _fetch_range(self, start, end):
+        pass
+
+
+class PrefixFileSystem(AbstractFileSystem):
+    """A meta-filesystem to add a prefix and delegate to another filesystem"""
+
+    def __init__(
+        self,
+        prefix: str,
+        fs: fsspec.AbstractFileSystem,
+        *args,
+        **storage_options,
+    ) -> None:
+        """
+        Parameters
+        ----------
+        prefix: str
+            The prefix to append to all paths
+
+        fs: fsspec.AbstractFileSystem
+            An instantiated filesystem to wrap. All operations are delegated to
+            this filesystem after appending the specified prefix
+        """
+        super().__init__(*args, **storage_options)
+        self.prefix = stringify_path(prefix)
+
+        if not self.prefix:
+            self.prefix = self.root_marker
```

Comment:
Continuing from https://github.com/intake/filesystem_spec/pull/745#discussion_r722297683

Is there a use case for this? I'm not sure this works right now, and doesn't seem like there are any tests for it. So maybe easier to just error out?

Reply:
The idea was to give the possibility to easily let the `PrefixFileSystem` be a passthrough that doesn't change any path